provider/cmd/pulumi-resource-gcp/main.go

```diff
@@ -19,15 +19,31 @@ package main
 import (
 	"context"
 	_ "embed"
+	"flag"
+	"io"
+	"os"
 
 	gcp "github.com/pulumi/pulumi-gcp/provider/v6"
 	"github.com/pulumi/pulumi-terraform-bridge/pf/tfbridge"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
 )
 
 //go:embed schema-embed.json
 var pulumiSchema []byte
 
+var tracing string
+
 func main() {
-	tfbridge.MainWithMuxer(context.Background(), "gcp",
-		gcp.Provider(), pulumiSchema)
+	module_name := "gcp"
+	flags := flag.NewFlagSet(module_name+"-flags", flag.ContinueOnError)
+	flags.StringVar(&tracing, "tracing", "", "Emit tracing to a Zipkin-compatible tracing endpoint")
+	flags.SetOutput(io.Discard)
+	err := flags.Parse(os.Args[1:])
+	contract.IgnoreError(err)
```

Comment:
It works fine to just ignore errors here because the tfbridge flags also understand tracing, but we'll have to be careful about adding flags that could conflict or that tfbridge doesn't understand and will error on. 

Maybe tfbridge should just expose the method for parsing its flags and we can optionally pass them in so it doesn't have to try to parse a second time

Reply:
Yeah, actually this is the third place we parse flags. There's one in `tf-bridge`, one in `pulumi/pulumi/pkg` and this one.  
I don't understand what you mean by `tfbridge should just expose the method for parsing its flags` - we still need to parse the flags and run the tracing init before passing to the `tf-bridge` code, so the duplicate parsing seems unavoidable.

Reply:
The parsing could be exported from tfbridge as a separate method. (so that it could be done before creating the provider and starting the bridge)

Reply:
Yes let's do that, or even make it called by default in tfbridge.MainWithMuxer and such, there's currently three entrypoints. Doing it this way will auto-install it across all providers, without us having to edit them.

Reply:
If I understand the code correctly, I don't think there's a way to auto-install the flag parsing without editing all providers.  

The control flow is `pulumi-gcp/main.go:main` -> `tf-bridge/MainWithMuxer` -> `pulumi/pulumi/pkg:Main`.  
We currently parse `--tracing` in `pulumi/pulumi/pkg:Main` and the `Provider` is initialized in `pulumi-gcp/main`, which is why tracing currently doesn't work before calling `pkg:Main`. We need to parse `--tracing` before initializing `Provider` in order to be able to trace it. I believe this necessarily means we need to edit `pulumi-gcp/main.go`.  

Also I think we'd want to make the changes so that we don't have to edit all providers at the same time (or should we?). Does the following then make sense:  

Add a `parseFlags` function in tfbridge which will be called in `pulumi-gcp:main` and in `tf-bridge/MainWithMuxer`(if we want to keep existing provider behaviour). There would also be a `handleParsedFlags` function which might be backed by a global state to make sure we only run it the first time. We'd need this to be separate since some of the flag handling logic requires that the `Provider` is already initialized.

The flag parsing logic is also duplicated between `pf` and `pkg` - should we refactor this too?